Backport: Fix a couple of heap overflow vulnerabilities, all "disputed"
severity
https://github.com/godotengine/godot/issues/123608
CVE-2026-110734
CVE-2026-110735
CVE-2026-110736

Index: thirdparty/README.md
--- thirdparty/README.md.orig
+++ thirdparty/README.md
@@ -791,6 +791,7 @@ Collection of single-file libraries used in Godot comp
   * Modifications: License included in header.
   * Patches:
     - `smaz-0001-write-string-warning.patch` (GH-8572)
+    - `smaz-0002-decompress-input-bounds.patch` ([GH-124701](https://github.com/godotengine/godot/pull/124701))
 - `smolv.{cpp,h}`
   * Upstream: https://github.com/aras-p/smol-v
   * Version: git (9dd54c379ac29fa148cb1b829bb939ba7381d8f4, 2024)
